docker/docker_test.go

```diff
@@ -0,0 +1,99 @@
+package docker
+
+import (
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"reflect"
+
+	"github.com/stretchr/testify/require"
+)
+
+// NOTES:
+// - Run "go test" to run tests
+// - Run "gocov test | gocov report" to report on test converage by file
+// - Run "gocov test | gocov annotate -" to report on all code and functions, those ,marked with "MISS" were never called
+//
+// or
+//
+// -- may be a good idea to change to output path to somewherelike /tmp
+// go test -coverprofile cover.out && go tool cover -html=cover.out -o cover.html
+//
+
+const (
+	path = "/webhooks"
+)
+
+var hook *Webhook
+
+func TestMain(m *testing.M) {
+
+	// setup
+	var err error
+	hook, err = New()
+	if err != nil {
+		log.Fatal(err)
+	}
+	os.Exit(m.Run())
+	// teardown
+}
+
+func newServer(handler http.HandlerFunc) *httptest.Server {
+	mux := http.NewServeMux()
+	mux.HandleFunc(path, handler)
+	return httptest.NewServer(mux)
+}
+
+func TestWebhooks(t *testing.T) {
+	assert := require.New(t)
+	tests := []struct {
+		name     string
+		event    Event
+		typ      interface{}
+		filename string
+		headers  http.Header
+	}{
+		{
+			name:     "BuildEvent",
+			event:    BuildEvent,
+			typ:      BuildPayload{},
+			filename: "../testdata/docker/docker_hub_build_notice.json",
+			headers: http.Header{
+				"X-Github-Event": []string{"commit_comment"},
```

Comment:
If this header can be removed that would be great! it's kind of misleading in this test.